src/pybind/mgr/orchestrator.py

```diff
@@ -1162,7 +1193,7 @@ def __init__(self, name, placement=None, count=None):
         self.name = name
 
         #: Count of service instances
-        self.count = 1 if count is None else count
+        self.count = self.placement.count if self.placement else 1  # for backwards-compatibility
```

Comment:
I'd completely remove this property.

Reply:
this is scattered in so many places in the rook and ansible orchestrator so that I'm quite hesitant to change it in so many places. I'd be inclined to have that as a separate PR, which makes testing this also quite a bit easier

Reply:
ok

Reply:
We need static type checking to make it possible to do these kind of refactorings. Changes like this are trivial in C++